beta-src/src/utils/map/drawArrowFunctional.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from "react";
```

Comment:
Cool! I can see why it was originally written imperatively... this is pretty tricky.